### DISCUSSION TITLE
how to put breakpoint in "injected files" to ease debugging ?

### DISCUSSION THREAD

**Discussion by jdeantoni** [eclipse-langium/langium#909] [Q&A] [answered]
Hey,

I'm new to typescript and Langium and I'm not able to bound the  breakpoints I put in the code that will be injected. Typically, to debug a scope provider.

The `sourceMaps:true` is correctly set but diagnostics from the debugger tells me that the map files are not loaded.  

I may be wrong but I suspect this is due to the DI mechanism.

How do classical Langium developers do to debug their code ?
thanks for your help...

**Comment by msujew** [comment, marked as answer]
Hey @jdeantoni,

the language server (which contains the code for the scope provider) is running in a separate process. While the normal `Run Extension` debug config starts the debugger for the extension, you'll have to run the `Attach to Language Server` debug configuration in addition to actually debug the language server.

**Reply by jdeantoni** [reply]
Thanks ! you saved my debugging session :)
